service/resource/configmap/update.go

```diff
@@ -2,12 +2,64 @@ package configmap
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
 )
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
+	currentStateConfigMap, ok := currentState.(*v1.ConfigMap)
+	if !ok {
+		return nil, nil, nil, configMapAssertionError
+	}
+
+	desiredStateConfigMap, ok := desiredState.(*v1.ConfigMap)
+	if !ok {
+		return nil, nil, nil, configMapAssertionError
+	}
+
+	// If the current or desired state configmaps are empty,
+	// perform no action.
+	if currentStateConfigMap == nil {
+		return nil, nil, nil, nil
+	}
+	if desiredStateConfigMap == nil {
+		return nil, nil, nil, nil
+	}
+
+	// If the current and desired state configmaps have different names or namespaces,
+	// something bad is going on, so error out.
+	if currentStateConfigMap.Name != desiredStateConfigMap.Name {
+		return nil, nil, nil, configMapWrongNameError
+	}
+	if currentStateConfigMap.Namespace != desiredStateConfigMap.Namespace {
+		return nil, nil, nil, configMapWrongNamespaceError
+	}
+
+	// If the current state does not match the desired state,
+	// return the desired state as update.
+	if currentStateConfigMap.Data[r.configMapKey] != desiredStateConfigMap.Data[r.configMapKey] {
+		return nil, desiredStateConfigMap, nil, nil
+	}
+
 	return nil, nil, nil, nil
```

Comment:
Does operatorkit have a guarantee that `ProcessUpdateState` is only called if `currentState` is not equal to `desiredState`? If so, then this should be an error.
Otherwise, this would be executed in the case `currentState` and `desiredState` are equal, which I wouldn't classify as an error.

Needs clarification.

Reply:
Yeah, I'd like some more knowledge here please sir <3

Reply:
IMO this has nothing to do with the framework. I just thought the code will never go here. When it does, then this is specific to this resource implementation. My personal feedback to that is that the implementation of this step is more complicated than it should be, which probably caused my confusion. When you think this is right that way, fine. I think it could be more straight forward like other resource implementations.